recipes/freealut/all/patches/freealut-1.1.0-conan-build.patch

```diff
@@ -0,0 +1,184 @@
+From 8ed31fc8a82926ce97d2f82ab6137edad9bfc76a Mon Sep 17 00:00:00 2001
+From: Lucas K Dal Castel <lucas.castel@ael.com.br>
+Date: Sat, 28 Oct 2023 12:34:03 -0300
+Subject: [PATCH] freealut-1.1.0-conan-build
+
+---
+ CMakeLists.txt    | 72 +++++++++++++++++------------------------------
+ include/AL/alut.h | 26 +++++++++--------
+ 2 files changed, 41 insertions(+), 57 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 640f35b..220f6fc 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -1,14 +1,16 @@
+ # cmake project file by Prakash Punnoor
+-CMAKE_MINIMUM_REQUIRED(VERSION 2.0)
++CMAKE_MINIMUM_REQUIRED(VERSION 3.15)
+ 
+ SET(CMAKE_MODULE_PATH "${CMAKE_SOURCE_DIR}/admin/CMakeModules")
+ PROJECT(Alut C)
++include(GNUInstallDirs)
++
+ SET(PACKAGE "freealut")
+ SET(PACKAGE_TARNAME "freealut")
+ SET(PACKAGE_NAME "freealut library")
+ SET(PACKAGE_MAJOR_VERSION "1")
+-SET(PACKAGE_MINOR_VERSION "0")
+-SET(PACKAGE_BUILD_VERSION "1")
++SET(PACKAGE_MINOR_VERSION "1")
++SET(PACKAGE_BUILD_VERSION "0")
+ SET(PACKAGE_VERSION "${PACKAGE_MAJOR_VERSION}.${PACKAGE_MINOR_VERSION}.${PACKAGE_BUILD_VERSION}")
+ SET(MAJOR_VERSION "0")
+ SET(MINOR_VERSION "0")
+@@ -36,14 +38,14 @@ SET(ALUT_SOURCES src/alutBufferData.c
+                  src/alutVersion.c
+                  src/alutWaveform.c)
+ 
+-INCLUDE_DIRECTORIES(${Alut_SOURCE_DIR}/include)
++INCLUDE_DIRECTORIES(${CMAKE_SOURCE_DIR}/include)
+ 
+ SET(ADD_WFLAGS "")
+ SET(ADD_CFLAGS "")
+ SET(ADD_LDFLAGS "")
+ SET(ADD_LIBS "")
+ 
+-OPTION(BUILD_STATIC    "build static library too"   ON)
++option(BUILD_SHARED_LIBS "Build using shared libraries" ON)
```

Comment:
Preferably, don't add this as an option (which converts it into a cache variable). That way you don't need to explicitly set it as a `tc.cache_variable`. CMakeToolchain sets it by default, but as a plain `tc.variable`.

Reply:
Since this PR: https://github.com/conan-io/conan/pull/12401 the conan toolchain sets it as a cache_variable, so it might be usefull if someone wants to build from source with option shared=false and then with option shared=true.

Reply:
still, no BUILD_SHARED_LIBS as an cmake option is not needed, as the toolchain sets it